assignment3.rb

```diff
@@ -0,0 +1,143 @@
+require 'csv'
+
+class Employee
+  attr_reader :name, :age, :department, :division, :designation
+
+  def initialize(name, age, department, division, designation)
+    @name = name
+    @age = age
+    @department = department
+    @division = division
+    @designation = designation
+  end
+
+  def to_csv_row
+    [name, age, department, division, designation]
+  end
+end
+
+# Subclasses for different divisions
+class DevelopmentEmployee < Employee; end
+class QAEmployee < Employee; end
+class DevopsEmployee < Employee; end
+
+class EmployeeFactory
+  def create_employee(name, age, department, division, designation)
+    case division
+    when 'Development'
+      DevelopmentEmployee.new(name, age, department, division, designation)
+    when 'QA'
+      QAEmployee.new(name, age, department, division, designation)
+    when 'Devops'
+      DevopsEmployee.new(name, age, department, division, designation)
+    else
+      Employee.new(name, age, department, division, designation)
+    end
+  end
+end
+
+class HR
+  attr_reader :employees
+
+  def initialize
+    @employees = []
+  end
+
+  def add_employee(name, age, department, division, designation)
+    employee = EmployeeFactory.new.create_employee(name, age, department, division, designation)
+    @employees << employee
+  end
+
+  def get_all_employees
+    @employees
+  end
+
+  def save_employees_to_csv
+    CSV.open('employees.csv', 'w') do |csv|
+      csv << ['Name', 'Age', 'Department', 'Division', 'Designation']
+      @employees.each do |employee|
+        csv << employee.to_csv_row
+      end
+    end
+  end
+end
+
+class Department
```

Comment:
Each class should be in a separate file.

Reply:
done chnges